packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,252 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, IHandlerParameters, ImperativeError, ProcessUtils, GuiResult,
+    TextUtils, IDiffOptions } from "@zowe/imperative";
+import { tmpdir } from "os";
+import { Download, Upload, IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
+import LocalfileDatasetHandler from "../compare/lf-ds/LocalfileDataset.handler";
+import LocalfileUssHandler from "../compare/lf-uss/LocalfileUss.handler";
+import { CompareBaseHelper } from "../compare/CompareBaseHelper";
+import { CliUtils } from "@zowe/imperative";
+import { existsSync, unlinkSync } from "fs";
+
+export class LocalFile {
+    path: string;
+    zosResp: IZosFilesResponse;
+}
+export enum Prompt {
+    useStash,
+    doneEditing,
+    continueToUpload
+}
+
+/**
+ * A shared utility class that uss and ds handlers use for local file editing.
+ * @export
+ * @class EditUtilities
+ */
+export class EditUtilities {
+    /**
+     * Builds a temp path where local file will be saved. If uss file, file name will be hashed
+     * to prevent any conflicts with file naming. A given filename will always result in the
+     * same unique file path.
+     * @param {IHandlerParameters} commandParameters - parameters supplied by args
+     * @returns {Promise<string>} - returns unique file path for temp file
+     * @memberof EditUtilities
+     */
+    public static async buildTempPath(commandParameters: IHandlerParameters): Promise<string>{
+        const ext = (commandParameters.arguments.extension ? commandParameters.arguments.extension : ".txt");
+        let fileName: string;
+        if (commandParameters.positionals.includes('uss')){
+            // Hash in a repeatable way if uss fileName (to get around any potential special characters in name)
+            fileName = commandParameters.arguments.file;
+            const crypto = require("crypto");
+            const hash = crypto.createHash('sha256').update(fileName).digest('hex');
+            return tmpdir() +"\\" + hash  + "." + ext;
+        }else{
```

Comment:
Should be able to remove the `else` branch here, since you will return early anyway if the logic runs in the `if` statement